BlueWizard/SpeechSynthesizer.mm

```diff
@@ -68,36 +75,66 @@ -(Buffer *)bufferFor:(NSArray *)wrappedSamples {
 }
 
 -(void)fillBuffer:(NSMutableArray *)samples {
-    unsigned int frames = _tms5220->m_fifo_count;
-    short int buffer[frames];
-    _tms5220->process(buffer, frames);
+    short int buffer[SAMPLES_PER_HALF_FRAME];
+    _tms5220->process(buffer, SAMPLES_PER_HALF_FRAME);
     float scale = 1.0f / (1 << 15);
-    for (int i = 0; i < frames; i++) {
+    for (int i = 0; i < SAMPLES_PER_HALF_FRAME; i++) {
         [samples addObject:[NSNumber numberWithDouble:buffer[i] * scale]];
     }
 }
 
 -(void)writeData:(int)data {
-    _tms5220->wsq_w(1);
+    //_tms5220->wsq_w(1); // don't use these since we want to use the 5220 without worrying about timing the reads and writes
     _tms5220->data_w(data);
-    _tms5220->wsq_w(0);
+    //_tms5220->wsq_w(0);
+}
+
+-(int)readStatus {
+    int tms_status;
+    //_tms5220->rsq_w(1); // don't use these since we want to use the 5220 without worrying about timing the reads and writes
+    tms_status = _tms5220->status_r();
+    //_tms5220->rsq_w(0);
+    //NSLog(@"5220 DEBUG: Index: %lu, Status read: TS: %d, BL: %d, BE: %d", self.index, (tms_status&0x80)?1:0, (tms_status&0x40)?1:0, (tms_status&0x20)?1:0);
+    return tms_status;
 }
 
 -(void)speakFragment:(NSArray *)speechData
              samples:(NSMutableArray *)samples {
-    while (!_tms5220->m_ready_pin) {
-        [self fillBuffer:samples];
+    // If the chip hasn't started talking yet, and we have yet to send any data...
+    if ( !([self readStatus]&STATUS_TS_MASK) && (self.index == 0) ) {
```

Comment:
same here, I'd prefer if we:

    if ([self chipHasNotStartedTalkingYetAndSentNoData]) { ... }

and then have a method:

    -(BOOL)chipHasNotStartedTalkingYetAndSentNoData {
        return  !([self readStatus] & STATUS_TS_MASK) && !self.index;
    }